clippy_lints/src/disallowed_method.rs

```diff
@@ -36,23 +38,40 @@ declare_clippy_lint! {
     /// let mut xs = Vec::new(); // Vec::new is _not_ disallowed in the config.
     /// xs.push(123); // Vec::push is _not_ disallowed in the config.
     /// ```
+    ///
+    /// In the clippy.toml configuration, you can also give the reason for
+    /// disallowing methods:
+    /// ```toml
+    /// disallowed-methods = [
+    ///     { path = "std::vec::Vec::leak", reason = "no leaking memory" },
+    ///     # no reason for this one
+    ///     { path = "std::time::Instant::now" },
```

Comment:
Is it possible to mix these, so for example:

```toml
disallowed-methods = [
    { path = "std::vec::Vec::leak", reason = "no leaking memory" },
    "std::time::Instant::now",
]
```

or are the braces necessary when you specify a reason for something?

Reply:
The implementation seems like it would allow for that. @azdavis can you add to the existing test instead of creating a new directory in tests/ui-toml?

Reply:
sure